packages/x-date-pickers/src/internals/hooks/usePickerState.ts

```diff
@@ -175,15 +248,19 @@ export const usePickerState = <TInput, TDateValue>(
       onChange: handleInputChange,
       open: isOpen,
       rawValue: value,
-      openPicker: () => setIsOpen(true),
+      openPicker: () => {
```

Comment:
Need to test when controlling `props.open` and the user open the picker not through the input.
It will not reset the value, so the bug we are trying to solve will still appear.

This reset should probably be done in a `useEffect` when `isOpen` changes